activesupport/lib/active_support/testing/time_helpers.rb

```diff
@@ -11,7 +11,9 @@ class SimpleStubs # :nodoc:
       Stub = Struct.new(:object, :method_name, :original_method)
 
       def initialize
-        @stubs = Concurrent::Map.new { |h, k| h[k] = {} }
+        @stubs = Concurrent::Map.new do |h, k|
+          h.compute_if_absent(k) { {} }
+        end
```

Comment:
This is an odd one.  The `Concurrent::Map` was added in #24890, but I'm not clear about why.  The `@stubs` ivar is private (i.e. unique to the `SimpleStubs` instance), and each running test [gets its own `SimpleStubs` instance](https://github.com/rails/rails/blob/2497eb0d5daf4f9aebce692c5bfad3792fecc712/activesupport/lib/active_support/testing/time_helpers.rb#L242-L244).  And `travel_to` (the only [place where `simple_stubs` is used](https://github.com/rails/rails/blob/2497eb0d5daf4f9aebce692c5bfad3792fecc712/activesupport/lib/active_support/testing/time_helpers.rb#L169-L171)) is not thread-safe anyway.

It might be better to just convert this to a vanilla `Hash`:

```suggestion
        @stubs = Hash.new { |h, k| h[k] = {} }
```

Reply:
Just for the record, the concurrent hash is vanilla hash under cruby. My gut feeling was that it's thread-safe because of some edge-cases with `Thread.new` though I cannot confirm this hypothesis. Not an expert with Rails code :(

Reply:
> Just for the record, the concurrent hash is vanilla hash under cruby. My gut feeling was that it's thread-safe because of some edge-cases with `Thread.new` though I cannot confirm this hypothesis.

I'm not sure what you are suggesting by this.  Are you suggesting that we use a `Concurrent::Hash` instead...?

To clarify my previous comment, each test has its own individual `@stubs` instance, even when the tests are running in the same thread:

```ruby
# my_test.rb
require "active_support"
require "minitest/autorun"

class MyTest < ActiveSupport::TestCase
  test "foo" do
    pp ["foo", simple_stubs.object_id, simple_stubs.instance_variable_get(:@stubs).object_id]
  end

  test "bar" do
    pp ["bar", simple_stubs.object_id, simple_stubs.instance_variable_get(:@stubs).object_id]
  end
end
```

```bash
$ ruby my_test.rb
Run options: --seed 13813

# Running:

["foo", 760, 780]
.["bar", 800, 820]
.

Finished in 0.034992s, 57.1563 runs/s, 0.0000 assertions/s.

2 runs, 0 assertions, 0 failures, 0 errors, 0 skips
```

Therefore, a concurrent data structure is not necessary.

Furthermore, `SimpleStubs` + `@stubs` exist solely to support the `travel_to` method, which is, itself, not thread safe.  So using a concurrent data structure here is confusing, because it implies thread safety.  Therefore, it would be better to use a vanilla `Hash`.

Reply:
My suggestion was, that unless there is any underlying concurrency stuff, we can just ditch it for a regular hash. Let me do it...